File: exs/mundo_2/python/039.py
"""
Desafio 039

Problema: Faça um programa que leia o ano de nascimento de um jovem e informe,
          de acordo com a sua idade, se ele ainda vai se alistar ao serviço
          militar, se é a hora exata de se alistar ou se já passou do tempo
          do alistamento. Seu programa também deverá mostrar o tempo que falta
          ou que passou do prazo.

Resolução do problema:
"""
from datetime import date
ano_nasc = int(input('Informe o ano do seu nascimento: '))
ano_atual = date.today().year
idade = ano_atual - ano_nasc

if idade < 18:
    print('Ainda vai se ALISTAR.')
    print('Você vai se ALISTAR daqui {} ano(s), em {}'.format(18 - idade, ano_atual + (18 - idade)))
elif idade == 18:
    print('Está na hora de se ALISTAR.')
else:
    print('Passou da hora de se ALISTAR')
    print('Você deveria ter se ALISTADO há {} ano(s), em {}.'.format(idade - 18, ano_atual - (idade - 18)))

File: exs/mundo_2/java/039.java
/*
Desafio 038

Problema: Faça um programa que leia o ano de nascimento de um jovem e informe,
          de acordo com a sua idade, se ele ainda vai se alistar ao serviço
          militar, se é a hora exata de se alistar ou se já passou do tempo
          do alistamento. Seu programa também deverá mostrar o tempo que falta
          ou que passou do prazo.

Resolução do problema:
*/

import java.time.LocalDate;
import java.util.Date;
import java.util.Scanner;

class Main{
    public static void main(String[] args) {

        // instancia a classe Scanner
        Scanner scanner = new Scanner(System.in);

//        obtendo data da classe Date
//        Date d = new Date();
//        int anoAtual = d.getYear() + 1900;

        // obtendo data atual com LocalDate
        LocalDate hoje = LocalDate.now();
        int anoAtual = hoje.getYear();

        System.out.println("Digite o ano do seu nascimento abaixo: ");
        int anoNascimento = scanner.nextInt();

        int idade = anoAtual - anoNascimento;

        if (idade < 18 ) {
            System.out.printf("Você vai se alistar em %d ano(s), em %d", 18-idade, anoAtual + (18-idade));
        } else if (idade == 18) {
            System.out.println("Está na hora de se ALISTAR");
        } else System.out.printf("Passou da hora de se ALISTAR.\nVocê deveria ter se ALISTADO há %d ano(s), em %d.", idade-18, anoAtual - (idade-18));



    }
}